DatadogSessionReplay/Tests/Recorder/RecorderTests.swift

```diff
@@ -8,42 +8,32 @@ import XCTest
 @testable import DatadogSessionReplay
 @testable import TestUtilities
 
-private class RUMContextObserverMock: RUMContextObserver {
-    private var queue: Queue?
-    private var onNew: ((RUMContext?) -> Void)?
-
-    func observe(on queue: Queue, notify: @escaping (RUMContext?) -> Void) {
-        self.queue = queue
-        self.onNew = notify
-    }
-
-    func notify(rumContext: RUMContext?) {
-        queue?.run { self.onNew?(rumContext) }
-    }
-}
-
 class RecorderTests: XCTestCase {
+    private var recorder: Recorder?
```

Comment:
question/ Why this convention 🤔💭? Storing SUT on `XCTestCase` property is wrong for components that manage global state between their `init()` and `deinit()` (such would have to be orchestrated with `setUp()` and `tearDown()`). It is not the case for `Recorder` today, but we're introducing something that is not scalable.

Reply:
While taking the responsibilities away from the recorder it made this test of tests very implicit. This var was introduce to just hold the reference to the recorder otherwise tests were failing due to deallocation issues. I added tearDown to make sure we nullify SUT.